core/ipc/common/kafka/src/main/java/org/opennms/core/ipc/common/kafka/Utils.java

```diff
@@ -47,4 +47,15 @@
             Thread.currentThread().setContextClassLoader(classLoader);
         }
     }
+
+    public static <T> T runWithGivenClassLoader(final Supplier<T> supplier, ClassLoader classLoader) {
+        Objects.requireNonNull(supplier);
```

Comment:
Can we use `runWithNullcontextClassLoader` use this method as well, instead of duplicating the code (more or less) here?

Reply:
yup, makes sense.